File: to-do-app-front/src/app/model/task.ts
import { Project } from "./project";

export interface Task {
  id?: string | number,
  name: string,
  userId: number,
  description: string,
  dueDate?: Date,
  priority?: number, // 1, 2, 3
  reminders?: Date[],
  tags?: string[],
  completed: boolean,
  project?: Project
}

File: to-do-app/tasks-service/src/main/java/com/app/tasksservice/model/Task.java
package com.app.tasksservice.model;

import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Entity(name = "TASKS")
@Data
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Long userId;

    @Column(nullable = false)
    private String description;

    private Date dueDate;

    private Integer priority; // 1, 2, 3

    @Column(nullable = false)
    private Boolean completed;

    private Long projectId;
}
